src/mappings/common.ts

```diff
@@ -79,3 +83,31 @@ export function exportFeeFromDepositEventAsString(extrinsic?: SubstrateExtrinsic
         return "0"
     } 
 }
+
+let currentEraByBlockId: {[blockId: string]: EraIndex} = {}
+
+export async function cachedCurrentEra(block: SubstrateBlock): Promise<EraIndex> {
+    let key = block.block.header.number.toString()
+    let cachedValue = currentEraByBlockId[key]
+    if (cachedValue !== undefined) {
+        return cachedValue
+    } else {
+        let eraOption = await api.query.staking.currentEra()
+        let eraIndex = eraOption.unwrap()
+        currentEraByBlockId[key] = eraIndex
+        return eraIndex
+    }
+}
+
+let eraStakersByEra: {[era: number]: [StorageKey<[EraIndex, AccountId]>, Exposure][]} = {}
+
+export async function cachedEraStakers(era: number): Promise<[StorageKey<[EraIndex, AccountId]>, Exposure][]> {
+    let cachedValue = eraStakersByEra[era]
+    if (cachedValue !== undefined) {
+        return cachedValue
+    } else {
+        let eraStakers = await api.query.staking.erasStakers.entries(era);
```

Comment:
I think we query and save this guys also when election comlpetes. So one improvement in future is to reuse that data.